testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1172,6 +1180,27 @@ public void testRenameColumn()
         super.testRenameColumn();
     }
 
+    @Test
+    public void testInsertNegativeDate()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT));
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "insert_date", "(dt DATE)")) {
+            if (supportsInsertNegativeDate()) {
+                assertUpdate(format("INSERT INTO %s VALUES (DATE '-2016-12-07')", table.getName()), 1);
+                assertQuery("SELECT * FROM " + table.getName(), "VALUES DATE '-2016-12-07'");
+            }
+            else {
+                assertQueryFails(format("INSERT INTO %s VALUES (DATE '-2016-12-07')", table.getName()), "(?s).*Failed to insert data.*");
+            }
+        }
+    }
+
+    protected boolean supportsInsertNegativeDate()
```

Comment:
`return hasBehaviour(XXX)`?

Reply:
@hashhar  I'm not sure if we want to reuse the enum for remote database limitation because existing values are for connectors implementation.

Reply:
Oh, that's a good point. This means that for some connectors we'll always have the limitation that we can't insert negative values but we should still be able to apply predicates with negative dates?

Reply:
@hashhar Changed to enum & `hasBehaviour` style. Please take another look.